### ISSUE TITLE
roster auto-add feature in salt-ssh-2018.3.0

### ISSUE THREAD

**Issue by edgan** [saltstack/salt#47532]
### Description of Issue/Question
I don't want this feature, and if it should stay it should be disabled by default.

```salt/client/ssh/__init__.py```:
```
                    roster_fp.write('# Automatically added by "{s_user}" at {s_time}\n{hostname}:\n    host: '
                                    '{hostname}\n    user: {user}'
                                    '\n    passwd: {passwd}\n'.format(s_user=getpass.getuser(),
                                                                      s_time=datetime.datetime.utcnow().isoformat(),
                                                                      hostname=self.opts.get('tgt', ''),
                                                                      user=self.opts.get('ssh_user', ''),
                                                                      passwd=self.opts.get('ssh_passwd', '')))
                log.info('The host {0} has been added to the roster {1}'.format(self.opts.get('tgt', ''),
                                                                                roster_file))
        else:
log.error('Unable to update roster {0}: access denied'.format(roster_file))
```

It causes things like in my roster:
```
# Automatically added by "edgan" at 2018-05-07T22:18:44.047773
None:
    host: None
    user: ubuntu
    passwd: 
```

Even if the None worked right, which is another bug, I don't want it editing my roster file.

### Setup
roster:
```
dns-dns-01.inf.us-west-2.aws.acme.com:
  host: 10.200.32.10
  sudo: True
```

### Steps to Reproduce Issue
```salt-ssh -i dns-dns-02.inf.us-west-2.aws.acme.com state.highstate```

### Versions Report
```
Salt Version:
           Salt: 2018.3.0
 
Dependency Versions:
           cffi: 1.11.2
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.5.5
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.14 (default, Mar 14 2018, 16:45:33)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.1.6
 
System Versions:
           dist: fedora 28 Twenty Eight
         locale: UTF-8
        machine: x86_64
        release: 4.16.6-302.fc28.x86_64
         system: Linux
        version: Fedora 28 Twenty Eight
```

**Comment by Ch3LL** [comment]
looks like you can disable this feature by setting `ssh_skip_roster`. It only runs `_update_roster` in this condition: `if not self.opts.get('ssh_skip_roster'):`  just noting this as a current workaround. Should be easy enough to disable by default.

ping @saltstack/team-ssh thoughts/opinions here on disabling this feature by default?

**Comment by gtmanfred** [comment]
I agree it should be disabled by default, and we should probably try and get that in for .1

**Comment by isbm** [comment]
`salt-ssh --skip-roster -i dns-dns-02.inf.us-west-2.aws.acme.com state.highstate <ENTER>`

**Comment by isbm** [comment]
The `None` thing is a bug indeed and needs to be fixed.

**Comment by rallytime** [comment]
Based on the discussion in #47541, I am bumping the severity of this issue down, as well as removing the `Blocker` label.

The `None` bug mentioned above has already been fixed with #47191. We can continue the discussion of the default behavior more carefully.

**Comment by rallytime** [comment]
Closed via #475441
